src/main/java/ir/sahab/nexus/plugin/tag/internal/TagRestResource.java

```diff
@@ -0,0 +1,150 @@
+package ir.sahab.nexus.plugin.tag.internal;
+
+import static org.sonatype.nexus.rest.APIConstants.V1_API_PREFIX;
+
+import ir.sahab.nexus.plugin.tag.internal.dto.Tag;
+import ir.sahab.nexus.plugin.tag.internal.dto.TagCloneRequest;
+import ir.sahab.nexus.plugin.tag.internal.dto.TagDefinition;
+import ir.sahab.nexus.plugin.tag.internal.exception.ErrorResponse;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validator;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.rest.Resource;
+
+/**
+ * Endpoint which provides RESTful API for tagging.
+ */
+@Named
+@Singleton
+@Path(V1_API_PREFIX + "/tags")
+public class TagRestResource extends ComponentSupport implements Resource, TagRestResourceDoc {
```

Comment:
Why not having an endpoint that gets a tag and a component and adds that component to tag's components list??
i think this end point is much useful.

Reply:
Why do u think it's useful? At the moment we create a tag for each stage and know all of its assocaited components at the time of tag creation, so why we should have it know? I know if we see it as a general tool which can be used by everyone we can have such endpoints as well, but I just wanted to stick to current requirements at the moment and keep those features based on future feedbacks.